### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by vkagamlyk**
*Issue #, if available:*
https://bitquill.atlassian.net/browse/AN-988

*Description of changes:*
Documentation update

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: gremlin-go/example/go.sum [hunk 1]
```diff
@@ -0,0 +1,22 @@
+github.com/BurntSushi/toml v0.3.1 h1:WXkYYl6Yr3qBf1K79EBnL4mak0OimBfB0XUf9Vl28OQ=
+github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
+github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c=
+github.com/google/uuid v1.3.0 h1:t6JiXgmwXMjEs8VusXIJk2BXHsn+wx8BZdTaoZ5fu7I=
+github.com/google/uuid v1.3.0/go.mod h1:TIyPZe4MgqvfeYDBFedMoGGpEw/LqOeaOT+nhxU+yHo=
+github.com/gorilla/websocket v1.4.2 h1:+/TMaTYc4QFitKJxsQ7Yye35DkWvkdLcvGKqM+x0Ufc=
+github.com/gorilla/websocket v1.4.2/go.mod h1:YR8l580nyteQvAITg2hZ9XVh4b55+EU/adAjf1fMHhE=
+github.com/lyndonbauto/tinkerpop/gremlin-go/v3 v3.5.3-rc12 h1:62id3eBpn79wef663sf0LHecGbR28vGdKKLAG63kPVk=
+github.com/lyndonbauto/tinkerpop/gremlin-go/v3 v3.5.3-rc12/go.mod h1:3GYWNfYdr+2wpLwsdXUQftwrmhUFA+Vfq3KWXjhY6Fw=
+github.com/nicksnyder/go-i18n/v2 v2.1.2 h1:QHYxcUJnGHBaq7XbvgunmZ2Pn0focXFqTD61CkH146c=
```
Comment (lyndonbauto) [inline gremlin-go/example/go.sum]: Needs to be apache/tinkerpop

File: gremlin-go/example/go.mod [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+module example
+
+go 1.17
+
+require github.com/lyndonbauto/tinkerpop/gremlin-go/v3 v3.5.3-rc12
```
Comment (lyndonbauto) [inline gremlin-go/example/go.mod]: Needs to be apache/tinkerpop

File: gremlin-go/example/go.sum (hunk 1 above)
Comment (vkagamlyk) [reply gremlin-go/example/go.sum]: updated in https://github.com/lyndonbauto/tinkerpop/pull/94/commits/e2e86986e299552753eab71559aa9cb7fd18643d

File: gremlin-go/example/go.mod (hunk 2 above)
Comment (vkagamlyk) [reply gremlin-go/example/go.mod]: updated in https://github.com/lyndonbauto/tinkerpop/pull/94/commits/e2e86986e299552753eab71559aa9cb7fd18643d

File: docs/src/reference/gremlin-variants.asciidoc [hunk 3]
```diff
@@ -1897,3 +1897,299 @@ pip install gremlinpython
...
+}
+----
+
+[[gremlin-go-lambda]]
+=== The Lambda Solution
+
+Supporting link:https://en.wikipedia.org/wiki/Anonymous_function[anonymous functions] across languages is difficult as
+most languages do not support lambda introspection and thus, code analysis. In Gremlin-Go, a Gremlin lambda should
+be represented as a zero-arg callable that returns a string representation of the lambda expected for use in the
+traversal. The lambda should be written as a `Gremlin-Groovy`string. When the lambda is represented in `Bytecode` its
```
Comment (lyndonbauto) [inline docs/src/reference/gremlin-variants.asciidoc]: Nit:  `Gremlin-Groovy`string appears to be missing a space.

Comment (vkagamlyk) [reply docs/src/reference/gremlin-variants.asciidoc]: fixed both https://github.com/lyndonbauto/tinkerpop/pull/94/commits/453b5b6f32cf87d27d618a1d968fe78ff26a4d6b

**Review by lyndonbauto — APPROVED** [review]

**Pull request merged**